vnext/Microsoft.ReactNative/Views/Image/ReactImage.cpp

```diff
@@ -128,6 +128,33 @@ winrt::IAsyncOperation<winrt::InMemoryRandomAccessStream> ReactImage::GetImageMe
       co_return nullptr;
   }
 }
+template <typename TImage>
+std::wstring GetUriFromImage(const TImage &image) {
+  return image.UriSource().ToString().c_str();
+}
+template <>
+std::wstring GetUriFromImage(const winrt::Uri &uri) {
+  return uri != nullptr ? uri.ToString().c_str() : L"<no Uri available>";
+}
+
+template <typename TImage>
+void ImageFailed(const TImage &image, const xaml::ExceptionRoutedEventArgs &args) {
+  cwdebug << L"Failed to load image " << GetUriFromImage(image) << L" (" << args.ErrorMessage().c_str() << L")"
+          << std::endl;
+}
+
+// TSourceFailedEventArgs can be either LoadedImageSourceLoadCompletedEventArgs or
+// SvgImageSourceFailedEventArgs, because they both have Status() properties
+// and the type of status are both enums with the same meaning
+// See LoadedImageSourceLoadStatus and SvgImageSourceLoadStatus.
+template <typename TImage, typename TSourceFailedEventArgs>
+void ImageFailed(const TImage &image, const TSourceFailedEventArgs &args) {
+  // https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.loadedimagesourceloadstatus
+  constexpr std::wstring_view statusNames[] = {L"Success", L"NetworkError", L"InvalidFormat", L"Other"};
+  const auto status = (int)args.Status();
+  assert(0 <= status && status < ARRAYSIZE(statusNames));
```

Comment:
We shouldn't be asserting when an image fails.   Images failing are not necessarily a code error on the part of RNW.

Reply:
that's not what the assert is saying - the assert is making sure the status is within the 4 defined values of  the LoadedImageSourceLoadStatus enum